Move attachments out from template to send_email

jxcl opened the pull request:
Defining attachments in the template did not allow for a template
to have a variable number of attachments. This patch allows templates
to be defined independently of the attachments they carry.

Commit:
Move attachments out from template to send_email

Defining attachments in the template did not allow for a template
to have a variable number of attachments. This patch allows templates
to be defined independently of the attachments they carry.

acatton commented on ogmios/__init__.py:
```diff
@@ -101,20 +106,16 @@ def get_recipients(self, name):
-                yield tuple(self.render_string(s) for s in attachment)
+    def validate_attachments(self):
+        for attachment in self.attachments:
+            if set(attachment.keys()) == set(['path', 'name', 'type']):
```
Why not support `{'path': ..., 'name': ...}` and a string. Passing a dict for just one  field seems complicated. And in the future we could support file-like objects.

acatton commented on CHANGELOG.rst:
```diff
@@ -1,10 +1,11 @@
 =========
 
-0.9.4 (unreleased)
+0.10.0 (unreleased)
```
It might be time to stabilize the API and release 1.0? What do you think?

jxcl replied:
1.0 should drop support for Django 1.7 since it's EOL and add support for 1.9.

Commit:
Allow more ways of specifying attachments

Attachments can now be specified with just a path to the filename
without a wrapping dict.
